### PR TITLE
[issue-1674] mixed nulls bug fix

### LINKED ISSUE #1674: [BUG] Mixed Nulls cannot deal with category dtypes
**Describe the bug**
Mixed Nulls check cannot work on "category" dtypes

**To Reproduce**
1. Create a dataframe with some columns with "category" dtypes
image
2. Then use mixed nulls check with default condition
image

**Expected behavior**
image

**Environment (please complete the following information):**
 - OS: Mac
 - Python Version:3.8
 - Deepchecks Version:0.8.0

**Additional context**

### PR DESCRIPTION
**Pull request by yromanyshyn**
resolves #1674

### CODE REVIEW

**Review by matanper — APPROVED** [review]

File: deepchecks/tabular/checks/data_integrity/mixed_nulls.py [hunk 1]
```diff
@@ -84,13 +86,18 @@ def run_logic(self, context: Context, dataset_kind) -> CheckResult:
         result_dict = {}
 
         for column_name in list(df.columns):
-            column_data = df[column_name]
-
-            string_null_counts = {value: count for value, count in column_data.value_counts(dropna=True).iteritems()
-                                  if string_baseform(value) in null_string_list}
+            column_data = (
```
Comment (matanper) [inline deepchecks/tabular/checks/data_integrity/mixed_nulls.py:89]: Add here a comment on why we cast categorical to object

Comment (yromanyshyn) [reply deepchecks/tabular/checks/data_integrity/mixed_nulls.py:89]: added

File: deepchecks/tabular/checks/data_integrity/mixed_nulls.py [hunk 2]
```diff
@@ -84,13 +86,26 @@ def run_logic(self, context: Context, dataset_kind) -> CheckResult:
...
+                # NOTE:
+                # `pandas.Series.value_counts` and 'pandas.Series.apply'
+                # work in an unusual way with categorical data types
+                # - `value_counts` returns all categorical values even if they are not in series
+                # - `apply` applies function to each category, not to values
+                # therefore we need cast it
+                # NOTE:
+                # `Series.astype('object')` will transform `numpy.nan` into `math.nan`
```
Comment (matanper) [inline deepchecks/tabular/checks/data_integrity/mixed_nulls.py:97]: I'm not sure we are okay with this, we want the output to represent the real nans of the user. As an alternative we just use `value_counts()` and filter all the values with 0

Comment (yromanyshyn) [reply deepchecks/tabular/checks/data_integrity/mixed_nulls.py]: agree (I changed implementation)
but `value_counts` also might transform `np.nan` into `math.nan`

image

Comment (yromanyshyn) [reply deepchecks/tabular/checks/data_integrity/mixed_nulls.py]: note: this does not happen with `name` column, at the end, it is `np.nan`

Comment (yromanyshyn) [reply deepchecks/tabular/checks/data_integrity/mixed_nulls.py]: whenever you try to iterate over the series, pandas transforms `np,nan` into `math.nan`

image

**Pull request merged**